### PR TITLE
AM2R: Add videos to DB, minor trick additions

### PR DESCRIPTION
**Pull request by JeffGainsNGames**
- 15 Videos
- IBJ to reach Inner Alpha Nest South Item
- Shinespark conservation to reach Hi-Jump Challenge Item

### CODE REVIEW

File: randovania/games/am2r/logic_database/Hydro Station.txt [hunk 1]
```diff
@@ -612,9 +612,12 @@ Extra - minimap_data: [{'x': 33, 'y': 23}, {'x': 34, 'y': 23}, {'x': 35, 'y': 23
                   Gravity Suit and Morph Ball and Speed Booster and Shinesparking Tricks (Beginner)
                   Any of the following:
                       Power Grip or Can Use Spring Ball
+                      # https://www.youtube.com/watch?v=mQahK4kIL3w
```
Comment (Miepee) [inline randovania/games/am2r/logic_database/Hydro Station.txt:615]: can you add another comment to the yt vid?

File: randovania/games/am2r/logic_database/Main Caves.txt [hunk 2]
```diff
@@ -497,6 +497,8 @@ Extra - minimap_data: [{'x': 49, 'y': 20}, {'x': 49, 'y': 21}, {'x': 50, 'y': 20
           Infinite Bomb Jumping (Advanced) and Can Use Bombs and Can Use Spider Ball
           # Diagonal IBJ
           Diagonal Infinite Bomb Jumping (Advanced) and Can Use Bombs
+          # Shinespark Up (Stored from previous room) https://www.youtube.com/watch?v=oMKgePFHxHY
+          Speed Booster and Shinesparking Tricks (Advanced) and Walljump (Intermediate) and Disabled Entrance Rando
```
[the remark this answers is not in the document]
Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/Main Caves.txt:501]: Honestly I'd vote to change this room to have HJ as the base requirement as SJ. I don't think there's any need for HJ to be locked by movement (int).
Agree bump to expert.

File: randovania/games/am2r/logic_database/The Tower.txt [hunk 3]
```diff
@@ -241,12 +241,12 @@ Extra - minimap_data: [{'x': 32, 'y': 27}, {'x': 32, 'y': 28}, {'x': 32, 'y': 29
                           Space Jump
                           Power Grip or Mid-Air Morph (Intermediate)
                       All of the following:
-                          # Shinespark Method
+                          # Shinespark Method https://www.youtube.com/watch?v=XSMA6Cr6O10
```
[the remark this answers is not in the document]
Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/The Tower.txt:244]: I always walked too far when walking off the cliff and then you only have a few moments to initiate the spark. Jumping allowed me more time to put the inputs in and was more consistent for me.

File: randovania/games/am2r/logic_database/The Tower.txt [hunk 4]
```diff
@@ -1316,6 +1316,7 @@ Extra - minimap_data: [{'x': 33, 'y': 46}, {'x': 34, 'y': 46}]
       All of the following:
           Super Missiles and Power Grip Wall
           Any of the following:
-              Knowledge (Intermediate)
               Speed Booster and Shinesparking Tricks (Beginner)
+              # https://www.youtube.com/watch?v=GiBGoj7z2Pg
```
[the remark this answers is not in the document]
Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/The Tower.txt:1320]: Tested, could not complete.

File: randovania/games/am2r/logic_database/Main Caves.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/am2r/logic_database/Main Caves.txt:501]: intermediate is probably a bit high. my argument for it being locked behind another trick, was that if you screw it up, you will get damaged a few times. does movement beginner sound like a nice compromise?

File: randovania/games/am2r/logic_database/The Tower.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/am2r/logic_database/The Tower.txt:1320]: :(

File: randovania/games/am2r/logic_database/The Tower.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/am2r/logic_database/The Tower.txt:244]: did you use aimlock for it?

File: randovania/games/am2r/logic_database/Hydro Station.txt (hunk 1 above)
Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/Hydro Station.txt:615]: What comment?

Comment (Miepee) [reply randovania/games/am2r/logic_database/Hydro Station.txt:615]: something like "mid air shinespark: [yt link]"

File: randovania/games/am2r/logic_database/Main Caves.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/Main Caves.txt:501]: I feel like if you are playing with that little HP for that to matter, it's on you if you fall.

[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/am2r/logic_database/Main Caves.txt:501]: maybe, but we also use movement beginner for stuff like the GT shaft near bombs where if you somehow fall down, you may not be able to get back up depending on your trick knowledge and items.

[the remark this answers is not in the document]
Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/Main Caves.txt:501]: That's a possible softlock vs "ouch 7dmg".
Seems weird the Hi-Jump Challenge requires trick with HJ.

File: randovania/games/am2r/logic_database/The Tower.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/The Tower.txt:244]: No, basically never...

[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/am2r/logic_database/The Tower.txt:244]: that'd probably do it.

File: randovania/games/am2r/logic_database/Main Caves.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/am2r/logic_database/Main Caves.txt:501]: ehhhhh
fine.

**Review by Miepee — APPROVED** [review]

**Pull request merged**